class-tgm-plugin-activation.php

```diff
@@ -828,22 +845,25 @@ public function notices() {
 								self_admin_url( 'plugin-install.php' )
 							);
 
-							$linked_plugin_groups[] = '<a href="' . esc_url( $url ) . '" class="thickbox">' . esc_html( $plugin_group_single_name ) . '</a>';
-						}
-						else {
-							$linked_plugin_groups[] = $plugin_group_single_name; // No hyperlink.
-						}
+							$linked_plugins[] = sprintf(
+								'<a href="%1$s" class="thickbox">%2$s</a>',
+								esc_url( $url ),
+								esc_html( $this->plugins[ $plugin_slug ]['name'] )
+							);
 
-						if ( isset( $linked_plugin_groups ) && (array) $linked_plugin_groups ) {
-							$plugin_groups = $linked_plugin_groups;
+						} else {
+							$linked_plugins[] = esc_html( $this->plugins[ $plugin_slug ]['name'] ); // No hyperlink.
 						}
 					}
+					unset( $plugin_slug, $url );
 
-					$last_plugin = array_pop( $plugin_groups ); // Pop off last name to prep for readability.
-					$imploded    = empty( $plugin_groups ) ? '<em>' . $last_plugin . '</em>' : '<em>' . ( implode( ', ', $plugin_groups ) . '</em> and <em>' . $last_plugin . '</em>' );
+					$count       = count( $plugin_group );
+					$last_plugin = array_pop( $linked_plugins ); // Pop off last name to prep for readability.
+					$imploded    = empty( $linked_plugins ) ? '<em>' . $last_plugin . '</em>' : '<em>' . ( implode( ', ', $linked_plugins ) . '</em> ' . esc_html__( 'and', 'tgmpa' ) . ' <em>' . $last_plugin . '</em>' );
```

Comment:
How about a small private/protected function that returns a string in `<em>`??

The comma between each plugin will be italicised, so any semantics the emphasis element is meant to have per plugin, is not applicable.

The `esc_html__()` should be `esc_html_x()` so that some context can be given to translators about what that "and" is for.